Mutex/functions/Test-Mutex.ps1

```diff
@@ -0,0 +1,54 @@
+﻿function Test-Mutex {
+    <#
+    .SYNOPSIS
+        Check if a mutex has been already used in another process.
+
+    .DESCRIPTION
+        Check if a mutex has been already used in another process.
+        This method can only check if a mutex by agiven name has been initialized in another process.
+        It does not check if the mutex is currently been locked.
+
+    .PARAMETER Name
+        Name of the mutex to acquire a lock on. If the mutex should be globally valid on a multi user system
+        prefix the name with "Global\". Otherwise it is assumed that it belongs only to the current
+        user session (equals the prefix "Local\").
+
+    .PARAMETER CheckIfLocked
+        If used Test-Mutex will also check if an existing Mutex has been already locked.
+
+    .EXAMPLE
+        PS C:\> Test-Mutex -Name MyModule.LogFile
+
+        Checks if the mutex 'MyModule.LogFile' has been created in another process.
+
+    .EXAMPLE
+        PS C:\> Test-Mutex -Name MyModule.LogFile -CheckIfLocked
+
+        Checks if the mutex 'MyModule.LogFile' has been created in another process and if it's locked.
+        Returns $false if it does not exist or if it's currently not locked.
+        Therefor: $false means you can use it/the corresponding ressource.
+    #>
+    [CmdletBinding()]
+    [OutputType([bool])]
+    Param (
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true)]
+        [string]
+        $Name,
+        [Parameter(Mandatory = $false, ValueFromPipeline = $false, ValueFromPipelineByPropertyName = $false)]
+        [switch]
+        $CheckIfLocked
+    )
+
+    process {
+            $existingMutex = $null
+            if ([System.Threading.Mutex]::TryOpenExisting($Name, [ref]$existingMutex) -eq $false) {
+                return $false
+            }
+            if ($CheckIfLocked) {
+                $canItBeLocked = $existingMutex.WaitOne(5)
+                if ($canItBeLocked) { $existingMutex.ReleaseMutex() }
+                return ($canItBeLocked -eq $false)
```

Comment:
Again a bit of a personal style thing - one I wouldn't bring up if I didn't ask for some other change anyway, sorry though - but I try to avoid comparing to boolean values - this could instead be:
`return -not $canItBeLocked`

Reply:
Changed it, next push will include it.